### PR TITLE
Rename promotion rules to conditions

### PR DESCRIPTION
**Pull request by mamhoff**
Promotion Rules are not rules, they are just conditions, as for a rule we need also some kind of consequence. This fixes the naming.

### CODE REVIEW

File: app/models/solidus_friendly_promotions/condition.rb [hunk 1]
```diff
@@ -34,7 +34,7 @@ def eligibility_errors
     end
 
     def to_partial_path
-      "solidus_friendly_promotions/admin/conditions/rules/#{model_name.element}"
+      "solidus_friendly_promotions/admin/conditions/conditions/#{model_name.element}"
```
Comment (tvdeyen) [inline app/models/solidus_friendly_promotions/condition.rb:37]: you might consider to use the same folder for the condition partials now

File: app/models/solidus_friendly_promotions/conditions/user.rb [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
-      include OrderLevelRule
+  module Conditions
+    class User < Condition
+      include OrderLevelCondition
 
-      has_many :promotion_rules_users,
-        class_name: "SolidusFriendlyPromotions::PromotionRulesUser",
-        foreign_key: :promotion_rule_id,
+      has_many :conditions_users,
+        class_name: "SolidusFriendlyPromotions::ConditionsUser",
```
Comment (tvdeyen) [inline app/models/solidus_friendly_promotions/conditions/user.rb:9]: maybe `ConditionUser` as join table name?

File: db/migrate/20240409093831_add_action_reference_to_rules.rb [hunk 3]
```diff
@@ -1,14 +1,15 @@
 class AddActionReferenceToRules < ActiveRecord::Migration[7.0]
+  LocalPromotionRule = SolidusFriendlyPromotions::Condition
```
Comment (tvdeyen) [inline db/migrate/20240409093831_add_action_reference_to_rules.rb:2]: is this reference really beneficial?

File: db/migrate/20240508080834_rename_promotion_rules_to_conditions.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+    rename_table :friendly_promotion_rules, :friendly_conditions
+    rename_table :friendly_promotion_rules_stores, :friendly_conditions_stores
+    rename_table :friendly_promotion_rules_taxons, :friendly_conditions_taxons
+    rename_table :friendly_promotion_rules_users, :friendly_conditions_users
+    rename_table :friendly_products_promotion_rules, :friendly_conditions_products
+    rename_column :friendly_conditions_stores, :promotion_rule_id, :condition_id
+    rename_column :friendly_conditions_taxons, :promotion_rule_id, :condition_id
+    rename_column :friendly_conditions_users, :promotion_rule_id, :condition_id
+    rename_column :friendly_conditions_products, :promotion_rule_id, :condition_id
+  end
```
Comment (tvdeyen) [inline db/migrate/20240508080834_rename_promotion_rules_to_conditions.rb:12]: do you need to rename the type column values as well?

Comment (mamhoff) [reply db/migrate/20240508080834_rename_promotion_rules_to_conditions.rb:18]: yes

**Pull request merged**